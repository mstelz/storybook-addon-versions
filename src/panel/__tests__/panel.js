import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Panel from '..';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../utils/config');

const location = {
  pathname: '/0.2.5/',
  hash: '',
  search: '',
};

describe('Panel', () => {
  it('renders correctly, no versions (dev true)', () => {
    const storybook = {
      getQueryParam: () => 'true',
      setQueryParams: () => {},
    };

    const tree = mount(<Panel storybook={storybook} location={location} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, no versions (dev false)', () => {
    const storybook = {
      getQueryParam: () => 'false',
      setQueryParams: () => {},
    };

    const tree = mount(<Panel storybook={storybook} location={location} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders versions (dev false)', async () => {
    const storybook = {
      getQueryParam: () => 'false',
      setQueryParams: () => {},
    };

    const wrapper = await shallow(<Panel storybook={storybook} location={location} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders versions (dev true)', async () => {
    const storybook = {
      getQueryParam: () => 'true',
      setQueryParams: () => {},
    };

    const wrapper = await shallow(<Panel storybook={storybook} location={location} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('toggles dev mode', async () => {
    const storybook = {
      getQueryParam: () => 'true',
      setQueryParams: () => {},
    };

    const wrapper = await shallow(<Panel storybook={storybook} location={location} />);
    const linksFound = wrapper.find('button').length;
    wrapper.find('#versionsAddonDevMode').simulate('change');
    expect(wrapper.find('button').length).toBe(linksFound - 1);
    wrapper.find('#versionsAddonDevMode').simulate('change');
    expect(wrapper.find('button').length).toBe(linksFound);
  });
});
