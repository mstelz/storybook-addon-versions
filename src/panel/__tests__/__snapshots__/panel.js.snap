// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly, no versions (dev false) 1`] = `ReactWrapper {}`;

exports[`Panel renders correctly, no versions (dev true) 1`] = `ReactWrapper {}`;
